rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/AbstractBrooklynRestResource.java

```diff
@@ -49,12 +50,18 @@
 
     
     private ManagementContext managementContext;
+    private ManagementContextInternal managementContextInternal;
```

Comment:
Karaf injects proxies to services, not the service objects themselves. This loses multiple inheritance of interfaces that act as services.

As a result, even if the bean from `brooklyn-core` implements both  `org.apache.brooklyn.api.mgmt.ManagementContext` and `org.apache.brooklyn.core.mgmt.internal.ManagementContextInternal`, the proxies for each service will only implement their respective interface, and cannot be cast.

Also the fact that two services are implemented by the same object is not the concern of the client of those services.